### PR TITLE
Define table size per scaling test 

### PR DESCRIPTION
**Pull request by jimmyzhai**
This pull request adds a new header file `defines.h` for constants definitions. It defines different table sizes for outbound routing and ca_to_pa mapping according to [hero-implementation-details](https://github.com/sonic-net/DASH/blob/main/documentation/general/program-scale-testing-requirements/hero-implementation-details.md).

The hardcoded table sizes of outbound routing and ca_to_pa mapping, whose size 4M and 8M respectively (from https://github.com/sonic-net/DASH/pull/592) take up too much memory by default. This memory pressure could make low-RAM KVM DPU crash and be unusable with sonic mgmt test. 

The below shows the dash pipeline memory usage of the container `simple_switch` with different scale number of routing and mapping.
```
1. 4M/8M 
junhuazhai@junhuazhai-dev-vm:~/workspace/DASH/dash-pipeline$ docker stats --no-stream simple_switch-junhuazhai
CONTAINER ID   NAME                       CPU %     MEM USAGE / LIMIT     MEM %     NET I/O   BLOCK I/O   PIDS
00016dc28662   simple_switch-junhuazhai   0.08%     1.799GiB / 62.78GiB   2.87%     0B / 0B   0B / 0B     20

2. 4K/8K
junhuazhai@junhuazhai-dev-vm:~/workspace/DASH/dash-pipeline$ docker stats --no-stream simple_switch-junhuazhai
CONTAINER ID   NAME                       CPU %     MEM USAGE / LIMIT     MEM %     NET I/O   BLOCK I/O   PIDS
bcb858280193   simple_switch-junhuazhai   0.08%     46.63MiB / 62.78GiB   0.07%     0B / 0B   0B / 0B     20
```

### CODE REVIEW

File: dash-pipeline/bmv2/defines.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef _DASH_DEFINES_H_
+#define _DASH_DEFINES_H_
+
+#ifdef TABLE_FULL_SCALE
```
Comment (r12f) [inline dash-pipeline/bmv2/defines.h:4]: will be better to use TABLE_MINI_HERO_SCALE

**Comment by r12f** [comment]
thanks a lot Junhua for making this change!

File: dash-pipeline/bmv2/defines.h (hunk 1 above)
Comment (mgheorghe) [reply dash-pipeline/bmv2/defines.h:4]: TABLE_BABY_HERO_SCALE i believe it is in the range of 80K to 320K can we at least have the default  limits increased to that
image

Comment (jimmyzhai) [reply dash-pipeline/bmv2/defines.h]: Use TABLE_HERO_SCALE/TABLE_BABY_HERO_SCALE for hero/baby-hero respectively.

**Review by r12f — APPROVED** [review]
LGTM, thanks for getting the numbers and macro align with the spec!

**Comment by r12f** [comment]
@mgheorghe , would you like to take another look?

**Pull request merged**